From 662eee43eab6d91ed0dfa2c8890e6097353b8b17 Mon Sep 17 00:00:00 2001
From: Lucas D Hedding <lucashedding@1463982.no-reply.drupal.org>
Date: Fri, 30 Aug 2013 16:17:50 -0500
Subject: [PATCH] Issue #1805238 by duvalbruno: Fixed reset() expects
 parameter 1 to be array, boolean given in reset() (line 208
 of commerce_price_components.module).

---
 commerce_price_components.module |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/commerce_price_components.module b/commerce_price_components.module
index 83e4895..0117bd9 100644
--- a/commerce_price_components.module
+++ b/commerce_price_components.module
@@ -205,7 +205,7 @@ function commerce_price_components_get_components($data, array $options, $name,
 
 function commerce_price_components_get_component($data, array $options, $name, $type, $info) {
   if (isset($name)) {
-    $data = reset($data);
+    $data = is_array($data) ? reset($data) : NULL;
     return isset($data['price'][$name]) ? $data['price'][$name] : FALSE;
   }
 }
-- 
1.7.10.4

